### PR TITLE
Update deps to typescript 5.4

### LINKED ISSUE #48: Update dependencies for Typescript 5.4

### PR DESCRIPTION
**Pull request by carlansley**
Fixes #48.  I didn't initially think this would be necessary, but I noticed a subtle bug to be fixed.

### CODE REVIEW

**Comment by carlansley** [comment]
@lukebrdn i assigned https://github.com/checkdigit/doc/issues/101 to you, the publish and publish-beta actions (for public repos) in the doc should look like the ones in this PR.

File: src/node/pipeline.ts [hunk 1]
```diff
@@ -392,8 +392,8 @@ export default function <Sink>(...argumentList: unknown[]): Promise<Sink | void>
    */
   if ((sink as Writable).writable && !(sink as Readable).readable) {
     return new Promise((resolve, reject) => {
-      stream.pipeline(...(argumentList as Parameters<typeof stream.pipeline>), (error: unknown) => {
-        if (error === undefined) {
+      stream.pipeline(...(argumentList as Parameters<typeof stream.pipeline>), (error?: Error | null) => {
+        if (error === undefined || error === null) {
```
Comment (carlansley) [inline src/node/pipeline.ts:396]: ^ subtle bug

**Review by ramaghanta — APPROVED** [review]
lftm

**Review by lukebrdn — APPROVED** [review]
lgtm

**Pull request merged**